scripts-dev/release.py

```diff
@@ -163,6 +165,8 @@ def prepare():
 
     # Switch to the release branch.
     parsed_new_version = version.parse(new_version)
+    # parse() shouldn't be returning a LegacyVersion. Whatever that is.
+    assert isinstance(parsed_new_version, version.Version)
```

Comment:
The docs suggest that we'll only see a `LegacyVersion` if we feed in non-PEP 440 compliant strings: https://packaging.pypa.io/en/latest/version.html#packaging.version.parse

Which ought to never happen with our version numbering, so the assert is fine!